addons/point_of_sale/models/report_sale_details.py

```diff
@@ -171,9 +171,11 @@ def get_sale_details(self, date_start=False, date_stop=False, config_ids=False,
                             payment['count'] = True
                         elif payment['id'] in account_payments.mapped('pos_payment_method_id.id'):
                             account_payment = account_payments.filtered(lambda p: p.pos_payment_method_id.id == payment['id'])
+                            profit_line = account_payment.move_id.line_ids.filtered(lambda l: l.account_id == account_payment.journal_id.profit_account_id)
+                            loss_line = account_payment.move_id.line_ids.filtered(lambda l: l.account_id == account_payment.journal_id.loss_account_id)
                             payment['final_count'] = payment['total']
-                            payment['money_counted'] = sum(account_payment.mapped('amount_signed'))
-                            payment['money_difference'] = payment['money_counted'] - payment['final_count']
+                            payment['money_difference'] = -loss_line.amount_currency if loss_line else profit_line.amount_currency
```

Comment:
I don't understand why there is a sign difference between profit and loss.
The difference should be one sign if it is a profit, and another sign if it is a loss right? Which is the case of `amount_currency` already